### PR TITLE
Made reactive controllers

### PR DESCRIPTION
**Pull request by ArSiu**

### CODE REVIEW

File: eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt [hunk 1]
```diff
@@ -18,9 +19,11 @@ class NatsControllerBeanPostProcessor : BeanPostProcessor {
     private fun <ReqT : GeneratedMessageV3, RepT : GeneratedMessageV3>
             NatsController<ReqT, RepT>.initializeNatsController() {
         connection.createDispatcher { message ->
-            val parsedData = parser.parseFrom(message.data)
-            val response = handler(parsedData)
-            connection.publish(message.replyTo, response.toByteArray())
+            handler(parser.parseFrom(message.data))
+                .subscribeOn(Schedulers.boundedElastic())
```
Comment (ajax-shulepa-a) [inline eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt]: double-check that map and doOnNext are running in your boundedElastic treads

File: eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt [hunk 2]
```diff
@@ -18,19 +20,32 @@ class AssignmentNatsControllerDeleteById(
...
+        val assignmentRequest = request.request
+
+        return when (assignmentRequest.requestCase) {
+            AssignmentRequest.RequestCase.ASSIGNMENT_ID ->
+                Mono.defer {
+                    val id = assignmentRequest.assignmentId.id
+                    service.delete(id).thenReturn(getSuccessResponse())
+                }.onErrorResume {
+                    Mono.just(getFailureResponse(it.javaClass.simpleName, it.toString()))
+                }
```
Comment (ajax-shulepa-a) [inline eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt]: extract this code to function, so when expression will be shorter

File: eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt [hunk 3]
```diff
@@ -18,19 +20,32 @@ class AssignmentNatsControllerDeleteById(
...
+
+            AssignmentRequest.RequestCase.ASSIGNMENT ->
+                Mono.just(getFailureResponse("UnsupportedRequestType", "AssignmentProto is not supported for deletion"))
+
+            AssignmentRequest.RequestCase.REQUEST_NOT_SET ->
+                Mono.just(getFailureResponse("InvalidRequest", "No request type set within AssignmentRequest"))
+
+            else ->
+                Mono.just(getFailureResponse("UnknownRequestType", "Unknown request type within AssignmentRequest"))
```
Comment (ajax-shulepa-a) [inline eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt]: let's handle all possible cases (without else branch). So, when somebody adds a new case to oneof, he is obliged by the compilator to add handling of his case. fix in all cases

File: eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerFindAll.kt [hunk 4]
```diff
@@ -20,11 +21,11 @@ class AssignmentNatsControllerFindAll(
...
     override val parser: Parser<FindAllAssignmentRequest> = FindAllAssignmentRequest.parser()
 
-    override fun handler(request: FindAllAssignmentRequest): FindAllAssignmentResponse =
-        runCatching {
-            getSuccessResponse()
-        }.getOrElse { ex ->
-            getFailureResponse(ex.javaClass.simpleName, ex.toString())
+    override fun handler(request: FindAllAssignmentRequest): Mono<FindAllAssignmentResponse> =
+        Mono.defer {
+            Mono.just(getSuccessResponse())
```
Comment (ajax-shulepa-a) [inline eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerFindAll.kt]: your getSuccessResponse uses `block` inside. so you don't have any async non-blocking features. refactor it to make truly reactive

**Review by ajax-shulepa-a — CHANGES_REQUESTED** [review]

File: eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt (hunk 1 above)
Comment (ArSiu) [reply eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt]: added .publishOn(Schedulers.boundedElastic())
So all other operators will exec at boundedElastic()

File: eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt (hunk 2 above)
Comment (ArSiu) [reply eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt]: `+`

File: eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt (hunk 3 above)
Comment (ArSiu) [reply eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt]: `+`

File: eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerFindAll.kt (hunk 4 above)
Comment (ArSiu) [reply eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerFindAll.kt]: `+`

File: eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt [hunk 5]
```diff
@@ -18,9 +19,22 @@ class NatsControllerBeanPostProcessor : BeanPostProcessor {
...
+                    .subscribeOn(Schedulers.boundedElastic())
+                    .publishOn(Schedulers.boundedElastic())
+                    .map { it.toByteArray() }
+                    .doOnNext { connection.publish(message.replyTo, it) }
+                    .subscribe()
+            }
         }.subscribe(subject)
     }
+
+
```
Comment (ajax-shulepa-a) [inline eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt]:
```suggestion
```

**Review by ajax-shulepa-a — APPROVED** [review]

File: eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt (hunk 5 above)
Comment (ArSiu) [reply eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt]: `+`

**Pull request merged**